### PR TITLE
topsql: fix data race in reporter test

### LINKED ISSUE #31251: DATA RACE in the ```reporter.(*collecting)```
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
==================
WARNING: DATA RACE
Read at 0x00c000b2d310 by goroutine 65:
  github.com/pingcap/tidb/util/topsql/reporter.(*collecting).getReportRecords()
      /go/tidb/util/topsql/reporter/datamodel.go:504 +0x1b8
  github.com/pingcap/tidb/util/topsql/reporter.(*RemoteTopSQLReporter).reportWorker()
      /go/tidb/util/topsql/reporter/reporter.go:280 +0x244
  github.com/pingcap/tidb/util/topsql/reporter.(*RemoteTopSQLReporter).Start·dwrap·5()
      /go/tidb/util/topsql/reporter/reporter.go:111 +0x39
Previous write at 0x00c000b2d310 by goroutine 61:
  github.com/pingcap/tidb/util/topsql/reporter.(*collecting).getOrCreateRecord()
      /go/tidb/util/topsql/reporter/datamodel.go:450 +0x167
  github.com/pingcap/tidb/util/topsql/reporter.(*RemoteTopSQLReporter).processCPUTimeData()
      /go/tidb/util/topsql/reporter/reporter.go:204 +0x399
  github.com/pingcap/tidb/util/topsql/reporter.populateCache()
      /go/tidb/util/topsql/reporter/reporter_test.go:58 +0x4ef
  github.com/pingcap/tidb/util/topsql/reporter.TestCollectAndSendBatch()
      /go/tidb/util/topsql/reporter/reporter_test.go:117 +0x185
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /usr/local/go/src/testing/testing.go:1306 +0x47
Goroutine 65 (running) created at:
  github.com/pingcap/tidb/util/topsql/reporter.(*RemoteTopSQLReporter).Start()
      /go/tidb/util/topsql/reporter/reporter.go:111 +0x118
  github.com/pingcap/tidb/util/topsql/reporter.setupRemoteTopSQLReporter()
      /go/tidb/util/topsql/reporter/reporter_test.go:96 +0x107
  github.com/pingcap/tidb/util/topsql/reporter.TestCollectAndSendBatch()
      /go/tidb/util/topsql/reporter/reporter_test.go:112 +0xfc
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /usr/local/go/src/testing/testing.go:1306 +0x47
Goroutine 61 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1504 +0x9d1
  go.uber.org/goleak.VerifyTestMain()
      /home/prow/go/pkg/mod/go.uber.org/goleak@v1.1.12/testmain.go:53 +0x59
  github.com/pingcap/tidb/util/topsql/reporter.TestMain()
      /go/tidb/util/topsql/reporter/main_test.go:26 +0x44
  main.main()
      _testmain.go:111 +0x264
==================
```

### PR DESCRIPTION
**Pull request by mornyx**
Signed-off-by: mornyx <mornyx.z@gmail.com>

Issue Number: close #31251

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
First, code coverage gets lower too much.

#31252 by @hawkingrei  65.7632% 
https://codecov.io/gh/pingcap/tidb/tree/271b962c5d79486ed80191095c9077d89701a27d/util

#31340   48.2142% 
https://codecov.io/gh/pingcap/tidb/tree/bad88dab045149362bc9ac99c323872faa99c1c9/util

**Comment by hawkingrei** [comment]
Please fix the lint.

```
[2022-01-06T03:39:36.204Z] util/topsql/reporter/single_target_test.go:62:18: ineffectual assignment to exist (ineffassign)

[2022-01-06T03:39:36.204Z] 	normalizedPlan, exist := server.GetPlanMetaByDigestBlocking([]byte("P1"), 5*time.Second)

```

**Comment by hawkingrei** [comment]
#31252 by @hawkingrei 

util/reporter/reporter.go   74.2857%   

https://codecov.io/gh/pingcap/tidb/tree/271b962c5d79486ed80191095c9077d89701a27d/util/topsql/reporter

#31340 

util/reporter/reporter.go   32.7586%  

https://codecov.io/gh/pingcap/tidb/tree/536ab99a99a0a4281fa4df5f7f4edbd3bdc8b647/util/topsql/reporter

File: util/topsql/reporter/single_target_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 PingCAP, Inc.
```
Comment (hawkingrei) [inline util/topsql/reporter/single_target_test.go:1]: This pr is to fix the data race. Why add a new test case for other components?

Comment (mornyx) [reply util/topsql/reporter/single_target_test.go:1]: Because the test coverage gets down while the data race is being repaired.

**Comment by mornyx** [comment]
> #31252 by @hawkingrei
> 
> util/reporter/reporter.go 74.2857%
> 
> https://codecov.io/gh/pingcap/tidb/tree/271b962c5d79486ed80191095c9077d89701a27d/util/topsql/reporter
> 
> #31340
> 
> util/reporter/reporter.go 32.7586%
> 
> https://codecov.io/gh/pingcap/tidb/tree/536ab99a99a0a4281fa4df5f7f4edbd3bdc8b647/util/topsql/reporter

I added a unit test to increase the coverage of `reporter.go` to 80%+

**Review by hawkingrei — APPROVED** [review]

**Review by crazycs520 — APPROVED** [review]

**Pull request merged**